hack/test/unit

```diff
@@ -18,16 +18,35 @@ TESTDIRS="${TESTDIRS:-./...}"
 exclude_paths='/vendor/|/integration'
 pkg_list=$(go list $TESTDIRS | grep -vE "($exclude_paths)")
 
-echo "${pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
+base_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings -v "/libnetwork" || :)
+libnetwork_pkg_list=$(echo "${pkg_list}" | grep --fixed-strings "/libnetwork" || :)
+
+echo "${libnetwork_pkg_list}" | grep --fixed-strings "libnetwork/drivers/bridge" \
 	&& if ! type docker-proxy; then
 		hack/make.sh binary-proxy install-proxy
 	fi
 
 mkdir -p bundles
-gotestsum --format=standard-quiet --jsonfile=bundles/go-test-report.json --junitfile=bundles/junit-report.xml -- \
-	"${BUILDFLAGS[@]}" \
-	-cover \
-	-coverprofile=bundles/profile.out \
-	-covermode=atomic \
-	${TESTFLAGS} \
-	${pkg_list}
+
+if [ "${base_pkg_list}" ]; then
```

Comment:
@tianon can correct me, but I think you want `-n` here

```suggestion
if [ -n "${base_pkg_list}" ]; then
```

Reply:
These are all identical:
```bash
#!/usr/bin/env bash

empty=''
nonempty='nonempty'

[ "${empty}" ] && echo "non-empty" || echo "empty"
[ -n "${empty}" ] && echo "non-empty" || echo "empty"
[ ! -z "${empty}" ] && echo "non-empty" || echo "empty"
[ "${empty}" != "" ] && echo "non-empty" || echo "empty"

[ "${nonempty}" ] && echo "non-empty" || echo "empty"
[ -n "${nonempty}" ] && echo "non-empty" || echo "empty"
[ ! -z "${nonempty}" ] && echo "non-empty" || echo "empty"
[ "${nonempty}" != "" ] && echo "non-empty" || echo "empty"
```

Produces:
```
empty
empty
empty
empty
non-empty
non-empty
non-empty
non-empty
```

```
$ bash --version                                              
GNU bash, version 4.4.23(1)-release (x86_64-unknown-linux-gnu)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Additionally, from `man test` (which is used by `[`):
```
       -n STRING
              the length of STRING is nonzero

       STRING equivalent to -n STRING
```

Both styles are used across repo, sometimes even mixed within one file. In `/hack`, for example:
https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/make.sh#L44
https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/make.sh#L67
https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/validate/swagger-gen#L14
https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/make/run#L31

https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/make.sh#L144
https://github.com/moby/moby/blob/656a5e2bdf8cc2be64fee1459821b7045b755b0d/hack/make/run#L35

Is there a guideline, which clearly defines how this is to be done?

Reply:
> Is there a guideline, which clearly defines how this is to be done?

Where possible, we try to use POSIX (and avoid "bashisms") for portability, e.g. to allow some scripts to be run in an environment where `bash` may not be present, or where `/bin/sh` is not an alias for `bash` (which it is in many environments). Referring to the posix docs; https://pubs.opengroup.org/onlinepubs/9699919799/utilities/test.html

> The two commands:
> 
> ```bash
> test "$1"
> test ! "$1"
> ```
>
> could not be used reliably on some historical systems. Unexpected results would occur if such a string expression were used and `$1` expanded to `'!'`, `'('`, or a known unary primary. Better constructs are:
> 
> ```bash
> test -n "$1"
> test -z "$1"
> ```

So from the above, looks like `-n` or `-z` is the recommended approach.